### PR TITLE
LPS-147920 Translate only filled fields

### PR DESCRIPTION
**Pull request by M-IgorFranca**
Related to this [Ticket](https://issues.liferay.com/browse/LPS-147920)

### CODE REVIEW

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
Just started reviewing :)

**Comment by naraandrade** [comment]
@M-IgorFranca There are some integration tests failing in this branch. Could you please take a look at it? You can run `yarn test` inside the module to see the results.

**Review by naraandrade — CHANGES_REQUESTED** [review]
@M-IgorFranca There are some integration tests failing in this branch. Could you please take a look at it? You can run `yarn test` inside the module to see the results.

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
Thanks @naraandrade , I didn't understand why my tests were failing. But now I know.

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js [hunk 1]
```diff
@@ -41,10 +41,18 @@ import populateNotificationsData from './util/populateNotificationsData';
 
 const deserializeUtil = new DeserializeUtil();
 
+function findEmptyElements(element, selectedLanguageId) {
+	if (element.data.label) {
+		if (!(selectedLanguageId in element.data.label)) {
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js:46]: @M-IgorFranca I think you can just add the condition of the second `if` to the first `if`, placing a `&&` before it.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js [hunk 2]
```diff
@@ -335,6 +345,31 @@ export default function DiagramBuilder() {
...
+				findEmptyElements(elements, selectedLanguageId)
+			);
+			if (!emptyLabel) {
+				setTranslations((previous) => {
+					return {...previous, [languageId]: definitionTitle};
+				});
+			}
+			else {
+				delete translations[languageId];
+				setTranslations(translations);
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js:362]: @M-IgorFranca Please, add a blank line between the update on translations and its usage.

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by naraandrade** [comment]
Great job @M-IgorFranca :) Just pointed some minor changes.

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:sf

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js (hunk 2 above)
Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js:361]: Off course. 👍🏾

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js (hunk 1 above)
Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/DiagramBuilder.js]: yes it makes sense.

**Comment by naraandrade** [comment]
@M-IgorFranca there's some formatting error happening with `DiagramBuilder.js`, please check it.

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:sf

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:test:relevant

**Review by naraandrade — CHANGES_REQUESTED** [review]
Hi @M-IgorFranca, I just noticed that your fix is not covering one case. When we enter the Single Approver definition and check the translations, all languages are marked as translated, but they only have the title translated. With your code, after selecting a language and then selecting other language, the previous one is updated to not-translated, but ideally, this update should be done just when we enter the definition. Could you take a look at this, please?

**Comment by michaelhashimoto** [comment]
ci:reevaluate:1307401_4310

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
> Hi @M-IgorFranca, I just noticed that your fix is not covering one case. When we enter the Single Approver definition and check the translations, all languages are marked as translated, but they only have the title translated. With your code, after selecting a language and then selecting other language, the previous one is updated to not-translated, but ideally, this update should be done just when we enter the definition. Could you take a look at this, please?

Done 👍🏾

**Comment by naraandrade** [comment]
Just started reviewing :)

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
ci:reevaluate:1307401_4420

**Comment by naraandrade** [comment]
@M-IgorFranca looks like there are some front-end tests failing. Could you take a look at it? Thanks :)

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
> @M-IgorFranca looks like there are some front-end tests failing. Could you take a look at it? Thanks :)

You were right, UpperToolbar yarn tests were broken. I managed to fix it by modifying the test mock. Thanks!

**Comment by M-IgorFranca** [comment]
ci:stop

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
@M-IgorFranca The failures are not related with your changes. You can send this PR directly to Brian.

**Comment by M-IgorFranca** [comment]
resent [here](https://github.com/brianchandotcom/liferay-portal/pull/124204)

**Pull request closed without merging**